lib/rubocop/cop/lint/unneeded_with_index.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for unneeded `with_index`.
+      #
+      # @example
+      #   # bad
+      #   ary.each_with_index do |v|
+      #     v
+      #   end
+      #
+      #   # good
+      #   ary.each do |v|
+      #     v
+      #   end
+      #
+      #   # bad
+      #   ary.each.with_index do |v|
+      #     v
+      #   end
+      #
+      #   # good
+      #   ary.each do |v|
+      #     v
+      #   end
+      #
+      class UnneededWithIndex < Cop
+        MSG_EACH_WITH_INDEX = 'Use `each` instead of `each_with_index`.'.freeze
+        MSG_WITH_INDEX = 'Remove unneeded `with_index`.'.freeze
```

Comment:
Normally you'd use `redundant` instead of `unneeded` in English.

Reply:
I changed it all together. This changes the name of this cop from `Lint/UnneededWithIndex` to `Lint/RedundantWithIndex`. 2cdcccf